x-pack/plugins/ml/common/types/storage.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EntityFieldType } from './anomalies';
+
+export const ML_ENTITY_FIELDS_CONFIG = 'ml.singleMetricViewer.partitionFields';
+
+export type PartitionFieldConfig =
+  | {
+      /**
+       * Relevant for jobs with enabled model plot.
+       * If true, entity values are based on records with anomalies.
+       * Otherwise aggregated from the model plot results.
+       */
+      anomalousOnly: boolean;
+      /**
+       * Relevant for jobs with disabled model plot.
+       * If true, entity values are filtered by the active time range.
+       * If false, the lists consist of the values from all existing records.
+       */
+      applyTimeRange: boolean;
+      sort: {
+        by: string; // 'anomaly_score' | 'name';
+        order: string; // 'asc' | 'desc';
```

Comment:
Should this type be updated with the commented code?

Reply:
Yes, updated in https://github.com/elastic/kibana/pull/81510/commits/c4203a3663302b63b996aab0d18f358e861ea559